core/src/main/scala/org/scalacheck/ops/GenFromConfig.scala

```diff
@@ -0,0 +1,150 @@
+      "the filters on this generator are too restrictive.")
+  def nextRandomIterator(rng: Random = Random): Iterator[T] = iteratorFromConfig(Seed(rng.nextLong()))
+
+  // Keeping the following methods around to keep source compatibility with 2.6.0 and avoid some of the pain
+  // of pulling in versions >= 2.7.0
+
+  /**
+   * An iterator made by [[Iterator.continually]] calling [[Gen.sample]].
+   *
+   * @note If the generator has filters, then this method could return None a lot. If you want a
+   *       safe way to filter the Nones, see [[iterator]]
+   *
+   * @note This will use a random seed for each sample. If you want to get the same iterator from
+   *       the same seed, you would have to use [[iterator]] and handle the possible.
+   *
+   * @return An iterator of Options which are None if the generator's filters ruled out the sample.
+   */
+  def sampleIterator: Iterator[Option[T]] = Iterator continually gen.sample
+
+  /**
+   * Converts this generator to an [[Iterator]] in the obvious (but potentially dangerous) way.
+   *
+   * @note This pulls an item from the generator one at a time, however, if the generator has too
+   * many restrictive filters, this can result in an infinite loop.
+   *
+   * @see [[iterator]] for a safer, bounded alternative.
+   *
+   * @return An [[Iterator]] that returns only the defined samples.
+   */
+  @deprecated("This is generally a bad idea and is easy enough to inline. It will be removed in the next major version.", "2.7.0")
+  def toUnboundedIterator: Iterator[T] = sampleIterator collect { case Some(x) => x }
+
+  /**
+   * Get a random element from this generator using the default seed.
+   *
+   * Generators can run out of samples and return empty results. Typically,
+   * this will be the result of bad Gen Parameters or having too many
+   * suchThat() restrictions that make it difficult to find the next sample.
+   */
+  @deprecated(
+    "If you need a random sample on each call, use .nextRandom()\n" +
+      "If you want a stateless function, use .head\n" +
+      "If you want to pass a specific Seed, use .configured(_.withSeed(seed)).head",
+    "2.3.0")
+  @throws[GenExceededRetryLimit](
+    "When the number of attempts to generate a valid sample is exhausted because " +
+      "the filters on this generator are too restrictive.")
+  def getOrThrow: T = headFromConfig(GenConfig.default.withSeed(Seed.random()))
+
+  /**
+   * @see [[head]]
+   */
+  @deprecated("Use .head instead. This method is contradictory " +
+    "(throwing an exception makes the function partially undefined and thus impure) and verbose " +
+    "(this has very similar semantics to the familiar .head method on Scala collections)", "2.7.0")
+  @throws[GenExceededRetryLimit](
+    "When the number of attempts to generate a valid sample is exhausted because " +
+      "the filters on this generator are too restrictive.")
+  def getOrThrowPure: T = head
```

Comment:
This is just to retain source code compatibility. Once I release the next major version, all of this stuff will be removed.

Reply:
Can we put a ticket for these things?  

It could: 
* document "major version" for which this can be removed, 
* be a reminder for getting the work scheduled.
* provide some easy clean-up work that could fill out a sprint or something.